pycbc/distributions/__init__.py

```diff
@@ -70,3 +72,173 @@ def read_distributions_from_config(cp, section="prior"):
         else:
             raise ValueError("Same parameter in more than one distribution.")
     return dists
+
+class JointDistribution(object):
+    """
+    Callable class that calculates the joint distribution built from a set of
+    distributions.
+
+    Parameters
+    ----------
+    variable_args : list
+        A list of strings that contain the names of the variable parameters and
+        the order they are expected when the class is called.
+    \*distributions :
+        The rest of the arguments must be instances of distributions describing
+        the individual distributions on the variable parameters.
+        A single distribution may contain
+        multiple parameters. The set of all params across the distributions
+        (retrieved from the distributions' params attribute) must be the same
+        as the set of variable_args provided.
+    \*\*kwargs :
+        Valid keyword arguments include `constraints`. `constraints` is a list
+        functions that accept a dict of parameters with the parameter name as
+        the key. If the constraint is satisfied the function should return
+        True, if the constraint is violated, then the function should return
+        False.
+
+    Attributes
+    ----------
+    variable_args : tuple
+        The parameters expected when the evaluator is called.
+    distributions : list
+        The distributions for the parameters.
+    constraints : list
+        A list of functions to test if parameter values obey multi-dimensional
+        constraints.
+
+    Examples
+    --------
+    An example of creating a joint distribution with constraint that total mass must
+    be below 30.
+
+    >>> from pycbc.distributions import uniform, JointDistribution
+    >>> def mtotal_lt_30(params):
+        ...    return True if params["mass1"] + params["mass2"] < 30 else False
+    >>> mass_lim = (2, 50)
+    >>> uniform_prior = Uniform(mass1=mass_lim, mass2=mass_lim)
+    >>> prior_eval = JointDistribution(["mass1", "mass2"], uniform_prior,
+        ...                               constraints=[mtotal_lt_30])
+    >>> print prior_eval([20, 1])
+
+    """
+
```

Comment:
Could you add a `name` property here, to make it like the other distributions? Just call it `'joint'`.